### PR TITLE
@orta => Use Fastlane for automated deploys

### LINKED ISSUE #359: Fastlane Everything
[This](https://github.com/krausefx/fastlane) looks amazing. We've talked before about using CI to push betas – maybe just on tags? I'm sure we can modify our Makefile accordingly. Open to having a conversation here about deployment strategies. 

Closing #266 in favour of this.

### PR DESCRIPTION
**Pull request by ashfurrow**
Fixes #359.

Need to get the following up and running:

- [x] Tests running
- [x] OSS install running
- [x] Signing
- [x] Updates version/build numbers
- [x] Generates release notes from CHANGELOG
- [x] Deploys to Hockey
- [x] Post-deploy slack notifications
- [x] Tag the build
- [x] Remove Makefile
- [x] Migrate Travis config
- [x] Update README/docs

### CODE REVIEW

**Comment by orta** [comment]
nice work on the keys

**Comment by ashfurrow** [comment]
Thanks – took the extra time to get it "Orta" right.

**Comment by ashfurrow** [comment]
So this is ready to go; tests pass locally. 

screen shot 2015-04-01 at 10 07 35 pm

_Somehow_ changing from using a Makefile to Fastlane to invoke `xcodebuild` for tests broke Travis again. Getting the following error:

```
iPhoneSimulator: Unable to connect to "com.apple.instruments.deviceservice.lockdown" (Error Domain=com.apple.CoreSimulator.SimError Code=146 "Unable to lookup in current state: Shutdown" UserInfo=0x7fb04b468750 {NSLocalizedDescription=Unable to lookup in current state: Shutdown})
iPhoneSimulator: Timed out waiting 120 seconds for simulator to boot, current state is 1.
```

I am _so_ tired of these failing CI tests. 

Tomorrow, if I'm feeling adventurous, I might try reverting to _just_ using `.travis.yml` to run tests instead of going through any other tools ¯\_(ツ)_/¯ But that'll be in a separate PR.

File: docs/deployment.md [hunk 1]
```diff
@@ -1,5 +1,27 @@
+### Building for Deployment
+
+So, Xcode has some problems and will actually fail to compile the first time or two you try to build it (radars have been filed). To compensate for this, you need to run `fastlane build_for_deploy` until it works.
```
Comment (ashfurrow) [inline docs/deployment.md]: They're [working on it](https://twitter.com/jckarter/status/581507956769845250).

**Comment by ashfurrow** [comment]
It'd be awesome to have @KrauseFx give it a one-over, too.

File: Gemfile [hunk 2]
```diff
@@ -5,5 +5,5 @@ gem 'cocoapods', '0.36.1'
 gem 'cocoapods-keys', :git => 'https://github.com/orta/cocoapods-keys.git'
 
 gem 'xcpretty'
-gem 'shenzhen'
 gem 'sbconstants'
+gem 'fastlane', :git => 'https://github.com/ashfurrow/fastlane.git'
```
Comment (ashfurrow) [inline Gemfile]: Will update once [this](https://github.com/KrauseFx/fastlane/pull/168) is resolved.

**Comment by ashfurrow** [comment]
It's back to this, eh? 

screen shot 2015-04-01 at 10 19 36 pm

File: Gemfile (hunk 2 above)
Comment (KrauseFx) [inline Gemfile]: :+1:

**Comment by KrauseFx** [comment]
This looks pretty cool! :+1: I'm not yet sold on the changelog part. Maybe you should use `YAML` or a different system to store them?

Once this is in production, I'd love to link to eidolon on the `fastlane` project page.

File: .travis.yml [hunk 3]
```diff
@@ -9,14 +9,14 @@ before_install:
   - chmod 600 ~/.netrc
 install: 
   - bundle install
-  - make prepare_ci
-  - make bundle
+  - bundle exec pod repo add artsy https://github.com/artsy/Specs.git
```
Comment (orta) [inline .travis.yml]: don't think this is needed anymore

File: fastlane/Fastfile [hunk 4]
```diff
@@ -0,0 +1,119 @@
+# Customise this file, documentation can be found here:
```
Comment (orta) [inline fastlane/Fastfile]: can kill these comments

File: fastlane/Fastfile [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+  # Build
+  ipa({
+    workspace: "Kiosk.xcworkspace",
+    configuration: "Release",
+    scheme: "Kiosk",
+    destination: "build",
+    embed: "fastlane/Distribution_net.artsy.kiosk.beta.mobileprovision",
+    identity: "iPhone Distribution: ART SY INC",
+    'no-clean' => ' ',
```
Comment (orta) [inline fastlane/Fastfile]: this feels weird, could that be a `true` instead?

File: fastlane/Fastfile [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  })
+
+  # Push to Hockey
+  hockey({
+    api_token: ENV['HOCKEY_API_TOKEN'],
+    notes_type: 1, # Specifies notes are in markdown format
```
Comment (orta) [inline fastlane/Fastfile]: instead of a comment you could declare a named local var of 1

**Comment by orta** [comment]
None of my opinions are blockers. I'm happy to have this merged whenever.

File: fastlane/Fastfile (hunk 6 above)
Comment (ashfurrow) [inline fastlane/Fastfile]: I like this.

Comment (ashfurrow) [inline fastlane/Fastfile]: Rather, I like your idea.

File: fastlane/Fastfile (hunk 5 above)
Comment (ashfurrow) [inline fastlane/Fastfile]: No, it'll be included in the command, so it needs to be empty.

File: fastlane/Fastfile (hunk 4 above)
Comment (ashfurrow) [inline fastlane/Fastfile]: coooool.

File: .travis.yml (hunk 3 above)
Comment (ashfurrow) [inline .travis.yml]: How so? Travis will need it for the fonts pod.

File: fastlane/Fastfile [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    success: true,        # optional, defaults to true
+    payload: {            # optional, lets you specify any number of your own Slack attachments
+      'Version' => version_number,
+      'What\'s new' => release_notes,
+    },
+    default_payloads: [],
+  )
+
+  # Tag release and push to GitHub
+  sh "cd .. ; git tag #{version_number} ; git push --tags"
```
Comment (ashfurrow) [inline fastlane/Fastfile]: There are Fastlane actions for these I should be using instead.

File: fastlane/Fastfile (hunk 5 above)
Comment (orta) [inline fastlane/Fastfile]: :+1:

Comment (ashfurrow) [inline fastlane/Fastfile]: Gonna [go with your idea instead](https://github.com/KrauseFx/fastlane/pull/168#issuecomment-88983393).

File: .travis.yml (hunk 3 above)
Comment (orta) [inline .travis.yml]: it's inside the `Podfile`, this is the old style

**Comment by orta** [comment]
:+1:

File: .travis.yml (hunk 3 above)
Comment (ashfurrow) [inline .travis.yml]: Oh, right.

**Comment by dblandin** [comment]
This is a great reference for setting up fastlane :+1:

File: fastlane/Fastfile [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    commands = keys.map { |key|
+      command = "bundle exec pod keys set #{key} '-'"
+      if key == keys.first
+        command += " Eidolon"
+      end
+
+      command
+    }.join(' ; ')
+
+    sh "cd .. ; #{commands} ; pod install"
```
Comment (KrauseFx) [inline fastlane/Fastfile]: We could use the built in [cocoapods action](https://github.com/KrauseFx/fastlane/blob/master/Actions.md#cocoapods) :+1:

File: fastlane/Fastfile (hunk 7 above)
Comment (KrauseFx) [inline fastlane/Fastfile]: Yes, you can use [push_to_git_remote](https://github.com/KrauseFx/fastlane/blob/master/Actions.md#push_to_git_remote). I haven't used it myself, but I heard good things about it :wink:

Comment (ashfurrow) [inline fastlane/Fastfile]: Yeah, didn't like that I couldn't specify my own tag; making a PR now.

File: fastlane/Fastfile (hunk 8 above)
Comment (ashfurrow) [inline fastlane/Fastfile]: Cooooool.

**Pull request merged**